arduino/libraries/librariesindex/index.go

```diff
@@ -125,7 +126,8 @@ func (idx *Index) FindRelease(ref *Reference) *Release {
 // FindIndexedLibrary search an indexed library that matches the provided
 // installed library or nil if not found
 func (idx *Index) FindIndexedLibrary(lib *libraries.Library) *Library {
-	return idx.Libraries[lib.Name]
+	clean := strings.ReplaceAll(lib.Name, "_", " ")
```

Comment:
Thanks for your pull request @stonehippo

Unfortunately this proposal will only transfer the bug from one format of library name to another. Consider what happens to the libraries with an underscore in their name:

```text
$ arduino-cli version
arduino-cli.exe  Version: git-snapshot Commit: 9f050282 Date: 2022-05-08T06:18:23Z

$ arduino-cli lib search Arduino_MKRENV
Downloading index: library_index.json.gz downloaded
Downloading index signature: library_index.json.sig downloaded
Name: "Arduino_MKRENV"
  Author: Arduino
  Maintainer: Arduino <info@arduino.cc>
  Sentence: Allows you to read the temperature, humidity, pressure, light and UV sensors of your MKR ENV Shield.
  Paragraph:
  Website: http://github.com/arduino-libraries/Arduino_MKRENV
  Category: Sensors
  Architecture: samd
  Types: Arduino
  Versions: [1.0.0, 1.1.0, 1.2.0, 1.2.1]
  Provides includes: Arduino_MKRENV.h

$ arduino-cli lib install Arduino_MKRENV@1.2.0
Downloading Arduino_MKRENV@1.2.0...
Arduino_MKRENV@1.2.0 downloaded
Installing Arduino_MKRENV@1.2.0...
Installed Arduino_MKRENV@1.2.0

$ arduino-cli lib list --updatable
No updates available.
```

The solution is to fix the root bug rather than trying to put a bandaid over it. Every library has a canonical name and the library must always be referred to by this name:

- If the library has [a `library.properties` metadata file](https://arduino.github.io/arduino-cli/dev/library-specification/#libraryproperties-file-format) (["1.5 format"](https://arduino.github.io/arduino-cli/dev/library-specification/#15-library-format-rev-22)), the library name is defined by the `name` field of `library.properties`.
- If the library does not have a `library.properties` metadata file (["1.0" or "legacy" format](https://arduino.github.io/arduino-cli/dev/library-specification/#old-library-format-pre-15)), the library name is defined by the folder name the library is installed in.

The bug is that some parts of Arduino CLI are using the library folder name as the library name even for "1.5 format" libraries, as is tracked here: https://github.com/arduino/arduino-cli/issues/932. That is the bug that must be fixed. https://github.com/arduino/arduino-cli/issues/1041 is only a symptom of https://github.com/arduino/arduino-cli/issues/932

Reply:
@per1234 fair point. Of course, based on the current library_index.json, the existing implementation harms more than the fix (about twice as many libs have spaces in the name as underscores). ;-)

I'm going to kill this, and go see if I can fix #932, since the lib upgrade function is less than ideal if more than 25% of the libraries get missed (1322 of the total 4828 libraries in the index have spaces in the name field that the CLI currently relies on; 776 contain spaces).

Reply:
@per1234 

After taking another look at #932, I've pushed a change to this PR. Rather than replacing the underscores in the installed name, it uses the `RealName` of the lib for upgrade lookups. If a library does not have a `library.properties` file, it is still listed, but cannot be upgraded via the `arduino-cli lib upgrade` command, since the index is currently relied upon find the latest version.

It might make sense to mark those libraries as "legacy" to declare that they have to be upgraded manually (or via the `install` command, if they're in the index but still not conforming to the 1.5 spec).

Note that the `uninstall` command uses a different resolving mechanism and will uninstall libraries based on their real names or directory names. It may also make sense to use this resolving mechanism in place of what I have here, but I'll leave that to reviewers to weigh in on.

Reply:
@per1234 ok, I noticed the tests I (somewhere stupidly) broke and fixed them. This is ready for review again when you have a moment.